lsp-mode.el

```diff
@@ -1519,6 +1519,25 @@ BODY should never return `t' value."
   download-in-progress?
   buffers)
 
+(defun lsp-clients-executable-find (find-command &rest args)
```

Comment:
nit: Can't this be named something more generic? This doesn't even reassemble the `execute-find`, both in parameters and implementation.

I would imagine that this should be named (since it's already inside `lsp-mode.el`) like `lsp-call-procecss` and be documented as a wrapper of `call-process` which returns output instead of inserting it in a buffer.
Same thing like how `lsp-async-start-process` exists.

With helper function like this, more generic name covers more use cases and keeps us from re-inventing the wheel.

Reply:
FTR there is `shell-command-to-string`

Reply:
> With helper function like this, more generic name covers more use cases and keeps us from re-inventing the wheel.

I think @kiennq have a good point here. Sometime I implemented a helper function that may work across all modules yet is hard for me to decide where to put it; especially, the function itself is for in very generic use. I wonder if there is a file that maybe we can place all our helper functions? (Sorry, I am quite new here. :P)

Reply:
> I would imagine that this should be named (since it's already inside lsp-mode.el) like lsp-call-procecss and be documented as a wrapper of call-process which returns output instead of inserting it in a buffer.
Same thing like how lsp-async-start-process exists.

Scope creep.

> FTR there is shell-command-to-string

You can't get the exit status with `shell-command-to-string`. Command errors still returns output from that function.

> if there is a file that maybe we can place all our helper functions?

I was wondering about the same thing. Every large project should have a utils file/module somewhere, but it's spread out all over the place in our case.